hudi-utilities/src/main/java/org/apache/hudi/utilities/UtilHelpers.java

```diff
@@ -152,8 +156,26 @@ public static InitialCheckPointProvider createInitialCheckpointProvider(
   }
 
   /**
-   *
+   * Factory method for creating custom DFSPathSelector.
    */
+  public static AbstractDFSPathSelector createSourceSelector(TypedProperties props, Configuration conf) {
```

Comment:
IMO we could keep this method inside the path related class if possible, so the code would be more straightforward.